CRM/Volunteer/BAO/Project.php

```diff
@@ -187,10 +187,10 @@ public static function supplyDefaults(array $params) {
     if (!array_key_exists('loc_block_id', $params)) {
       $params['loc_block_id'] = $defaults['loc_block_id'];
     }
-    if (!array_key_exists('profiles', $params)) {
+    if (!array_key_exists('profiles', $params) || !CRM_Volunteer_Permission::check('edit volunteer registration profiles')) {
```

Comment:
I have to raise a flag for mixing concerns. Is it a convenience to do it here? That is, the default is only used if the permission-check fails AND the parameter is not already set? One advantage of having a "single responsibility principle" function for permissions could be optimising the permissions look-up. Permissions checks could involve group-look-ups, no? I see it is a static function. Is there a singleton that caches the look-up?